### ISSUE TITLE
In-Domain GAN Inversion for Real Image Editing

### ISSUE THREAD

**Issue by hypknot74** [hypknot74/Paper-summaries#25]
### リンク
- arxiv: [https://arxiv.org/abs/2004.00049](https://arxiv.org/abs/2004.00049)
- project page: [https://genforce.github.io/idinvert/](https://genforce.github.io/idinvert/)
- code(tensorflow): [https://github.com/genforce/idinvert](https://github.com/genforce/idinvert)
- code(pytorch): [https://github.com/genforce/idinvert_pytorch](https://github.com/genforce/idinvert_pytorch)
- code (genforceというプロジェクトに組み込まれている): https://github.com/genforce/genforce

### 他参考サイト
* [18本目の論文: In-Domain GAN Inversion for Real Image Editing](https://qiita.com/wataoka/items/85a92ec66fb2432a9b4b#18%E6%9C%AC%E7%9B%AE%E3%81%AE%E8%AB%96%E6%96%87-in-domain-gan-inversion-for-real-image-editing)
* [【AI論文解説】画像を編集しやすいベクトルへ変換（IDInvertの詳細）](https://www.youtube.com/watch?v=O5lFoON3KtE) : Youtube動画

### 著者
Jiapeng Zhu, Yujun Shen, Deli Zhao, Bolei Zhou

### 投稿日付
Submitted on 31 Mar 2020 
ECCV2020

## 概要
通称IDInvert (or IDGI)

Encoder型と最適化型のGAN inversionを組み合わせたHybrid型のGAN inversionの新しい手法を提案。GAN inversionの精度を高めるだけでなく、その画像の操作を可能にするためにGANが学習した意味的な空間への写像を可能とするようにEncoderを学習させる。

## 手法

従来のGAN inversion(特に最適化型のGAN inversion)では高精度なinversionは可能でも、画像の操作は不安定という問題があった。これは、無理やり潜在変数に落とし込むように最適化することで、GANが学習した意味的な方向などを無視しているためである。これによってinversionした潜在変数が分布外になることもある。

### Domain-Guided Encoder

本手法でははじめにEncoderを用いて潜在変数を求める。
上: 従来手法　下: 提案手法
![image](https://user-images.githubusercontent.com/57225520/184801309-ebc07581-50a2-4720-82a4-cbafcd9ef0fa.png)

従来手法と異なる点は以下の3つ

1. 潜在変数同士(z_samとz_enc)を比較するのではなく、実画像と再構成画像(x_realとx_rec)を比較する。
    1. これにより、BackporpのパスにGが含まれ、Gが得た知識を踏まえたinversionが可能。また、潜在変数での比較よりも情報量が多い。例えば、異なるzから同じ画像が生成できる可能性はあるが、Encoderを使用すれば同じ画像からは同じzに埋め込まれ画像対潜在変数が1対1のような関係になるため、より整理された空間でのinversionが可能になる？
2. より実画像への応用を可能にするために、学習に生成画像ではなく実画像を使用する
3. 再構成画像の質を高めるために、再構成画像と実画像をDに入れてDも学習する。つまり、EとDを競わせる。

EとDの学習は以下のように表せる。
![image](https://user-images.githubusercontent.com/57225520/184801396-0efa26c1-7949-4cdb-9d38-f97f40d2a5da.png)
Eの学習ではperceptual loss(VGGの特徴量)も損失に加えている。Dの学習の最後の項はgradient penalty。
### Domain-Regularized Optimization

Encoderによる潜在変数を初期値にしたうえで、最適化型のGAN inversionを行う。
上: 従来手法　下: 提案手法
![image](https://user-images.githubusercontent.com/57225520/184801470-5bdccc60-5f30-43d8-9493-0ab3223c9cd9.png)
EncoderはGの意味的性質を理解していることから、Encoderによる潜在変数を損失関数の項として使用する(第3項)。
![image](https://user-images.githubusercontent.com/57225520/184801505-4c936678-9ee3-4720-b966-c87f0f422e7f.png)

## 結果
Image2StyleGANは最適化型のGAN inversion
![image](https://user-images.githubusercontent.com/57225520/184801537-399e1070-d4c0-48b2-ab70-afbc7ee469d7.png)
![image](https://user-images.githubusercontent.com/57225520/184801552-844314b2-ec90-4ea5-80bc-4fade135c589.png)
FFHQで学習した画像で顔以外の画像をinversionした結果。Image2StyelGANではうまく再構成できており、提案手法では顔のようなものが浮かび上がっている。これはImage2StyelGANが強いというよりも、提案手法のEncoderが顔画像に関する知識を持った上でinversionしているといえる。
![image](https://user-images.githubusercontent.com/57225520/184801574-531ce08d-d681-403f-ae5e-2d1222921132.png)
## Encoderの構造について

論文にはEncoderの具体的な構造が載っていなかった(と思う)が、[E2Style](https://www.notion.so/GAN-part1-350f560e466b4cfd84221de04e6a5fbc)の論文に載っていた。
![image](https://user-images.githubusercontent.com/57225520/184801642-9e90b14b-c502-43a5-9a5d-107671b53203.png)
